src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj

```diff
@@ -24,9 +24,8 @@
           BeforeTargets="GetPackageFiles">
 
     <ItemGroup Condition="'$(PackageTargetRuntime)' != ''">
-      <File Include="@(LibrariesRuntimeFiles)" TargetPath="framework" />
-      <!-- exclude shared libs as we only need static ones -->
-      <LibPackageExcludes Include="framework\%2A%2A\%2ANative$(LibSuffix)"/>
+      <!-- Include all libraries (managed, static, etc.), but exclude native shared libraries (those with LibSuffix) -->
+      <File Include="@(LibrariesRuntimeFiles)" TargetPath="framework" Condition="'%(LibrariesRuntimeFiles.IsNative)' != 'true' or '%(LibrariesRuntimeFiles.Extension)' != '$(LibSuffix)'" />
```

Comment:
this will still exclude .dll on Windows. it should be
```suggestion
      <File Include="@(LibrariesRuntimeFiles)" TargetPath="framework" Condition="!('%(LibrariesRuntimeFiles.IsNative)' == 'true' and '%(LibrariesRuntimeFiles.Extension)' == '$(LibSuffix)')" />
```

Reply:
It's the ame logic? https://www.wolframalpha.com/input?i=%21A+or+%21B mine is DNF yours is ANF

Reply:
It's the same logic? https://www.wolframalpha.com/input?i=%21A+or+%21B mine is DNF yours is ANF

Reply:
```xml
<Project>
 
   <ItemGroup>
    <LibrariesRuntimeFiles Include="libfile0.dll">
      <IsNative>true</IsNative>
    </LibrariesRuntimeFiles>
    <LibrariesRuntimeFiles Include="content.txt">
      <IsNative>true</IsNative>
    </LibrariesRuntimeFiles>
    <LibrariesRuntimeFiles Include="libfile2.dll">
      <IsNative>false</IsNative>
    </LibrariesRuntimeFiles>
    <LibrariesRuntimeFiles Include="libfile3.dll">
      <IsNative></IsNative>
    </LibrariesRuntimeFiles>
    <LibrariesRuntimeFiles Include="console.exe">
      <IsNative>true</IsNative>
    </LibrariesRuntimeFiles>
  </ItemGroup>


  <Target Name="DisplayFiles">
    <!-- Condition 1: Negation of AND -->
    <ItemGroup>
      <FilesWithCondition1 Include="@(LibrariesRuntimeFiles)" Condition="!('%(IsNative)' == 'true' and '%(Extension)' == '.dll')" />
    </ItemGroup>

    <!-- Condition 2: OR of Negations -->
    <ItemGroup>
      <FilesWithCondition2 Include="@(LibrariesRuntimeFiles)" Condition="'%(IsNative)' != 'true' or '%(Extension)' != '.dll'" />
    </ItemGroup>

    <Message Importance="High" Text="Files that meet Condition 1:" />
    <Message Importance="High" Text="@(FilesWithCondition1)" />
    <Message Importance="High" Text="Files that meet Condition 2:" />
    <Message Importance="High" Text="@(FilesWithCondition2)" />
  </Target>

</Project>
```
try running this `dotnet build repro.proj` to verify it produces the same results and that De Morgan's law holds. i prefer DNF because it's succinct.

aside: eng/liveBuilds.targets suggests IsNative metadata is always present, so we don't have to worry about HasMetadata condition.

Reply:
you're right, I got confused :)